src/Traits/GenericActionsTrait.php

```diff
@@ -257,4 +277,35 @@ public function fallback(string $path): ?Response
 
         return null;
     }
+
+    /**
+     * Download a media given its `uname`.
+     *
+     * @param string $uname Media `uname`.
+     * @param string|null $filename Original file name. If not provided or not updated, redirect to the correct URL.
+     * @return \Cake\Http\Response
+     */
+    public function download(string $uname, ?string $filename = null): Response
+    {
+        /** @var \BEdita\Core\Model\Entity\Media $media */
+        $media = $this->Objects->loadObject($uname, 'media', [], []);
+        if (empty($media->streams)) {
+            throw new NotFoundException();
+        }
+
+        $stream = $media->streams[0];
```

Comment:
meglio accedere con il `first`
```suggestion
        $stream = collection($media->streams)->first();
```